javascript/ql/experimental/adaptivethreatmodeling/test/endpoint_large_scale/ExtractEndpointDataInference.ql

```diff
@@ -0,0 +1,27 @@
+/*
+ * ExtractEndpointDataInference.ql
+ *
+ * This test surfaces the endpoints that pass the endpoint filters and have flow from a source for each query config,
+ * and are therefore used as candidates for scoring at inference time.
+ *
+ * This is equivalent to ExtractEndpointDataTraining.qlref, but testing the inference endpoints rather than the training
+ * endpoints. It detects CodeQL changes that impact the endpoints that get scored at inference time.
+ *
+ * This test does not actually score the endpoints and test for changes in the model predictions: that gets done in the
+ * integration tests.
+ */
+
+private import javascript as JS
+import extraction.NoFeaturizationRestrictionsConfig
+private import experimental.adaptivethreatmodeling.ATMConfig as AtmConfig
+private import experimental.adaptivethreatmodeling.NosqlInjectionATM as NosqlInjectionAtm
+private import experimental.adaptivethreatmodeling.SqlInjectionATM as SqlInjectionAtm
+private import experimental.adaptivethreatmodeling.TaintedPathATM as TaintedPathAtm
+private import experimental.adaptivethreatmodeling.XssATM as XssAtm
+private import experimental.adaptivethreatmodeling.XssThroughDomATM as XssThroughDomAtm
+
+query predicate isSinkCandidateForQuery(
+  AtmConfig::AtmConfig queryConfig, JS::DataFlow::PathNode sink
```

Comment:
```suggestion
  AtmConfig::AtmConfig queryConfig, JS::DataFlow::Node sink
```

Reply:
I want to keep the test as similar as possible to the actual extraction queries. The extraction queries use `DataFlow::PathNode` (e.g. `javascript/ql/experimental/adaptivethreatmodeling/src/SqlInjectionATM.ql`). I don't know if there's a reason they do this or not, but if we want to change those to `DataFlow::Node` (in which case we can change these as well), we should do so in a separate PR.

Reply:
The extraction queries use the PathNodes for a specific reason - namely, that the UI should be able to list a specific path from source to sink. This is not needed in this test.

Reply:
See my comment [here](https://github.com/github/codeql/pull/11532#discussion_r1038366721)